### PR TITLE
Rdr 28 search results show only latest work versions

### PR DESCRIPTION
**Pull request by ad96**
info on searching works in RDR and the 'latest version' filter this PR is adding:

1) When a user does a search using the upper page search box (labelled "Go"), the "latest_version" filter will automatically apply, due to a hidden param that is included with that search -- whether it is an empty search or with search terms, only latest versions of works will be in search results.

2) The latest versions filter will not apply to "Browse Datasets" search or to any of the searches or links done specific to the "My Dashboard" or "My Works" searches.

3) 'latest_version' work is any work that does not have a value in the metadata field 'is_replaced_by'.

### CODE REVIEW

File: spec/controllers/catalog_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      let!(:v1) { FactoryBot.create(:dataset, :public, doi: "http://example.com/my_doi_v1", is_replaced_by: "http://example.com/my_doi_v2") }
+      let!(:v2) { FactoryBot.create(:dataset, :public, doi: "http://example.com/my_doi_v2", replaces: "http://example.com/my_doi_v1") }
+      it "retrieves only the latest version" do
+        get :index
+        expect(assigns(:document_list).map(&:id)).to match_array [v2.id, v1.id]
+      end
+    end
+  end
+end
+
```
Comment (coblej) [inline spec/controllers/catalog_controller_spec.rb]: Not a big deal but I would remove these extra lines (or at least all but one of them) from the end of this file.

Comment (coblej) [reply spec/controllers/catalog_controller_spec.rb]: Meaning the black lines at the end of the file.

Comment (ad96) [reply spec/controllers/catalog_controller_spec.rb]: Thanks! I've removed the extra blank lines.

**Comment by seanaery** [comment]
The code looks good to me, though I would definitely squash these 11 commits down to one before merging. @ayse I'm happy to help with that if you'd like.

**Comment by ad96** [comment]
Yes, @seanaery we will definitely squash all the commits for this change. Does the 'Squash and merge' button take care of it?

**Comment by seanaery** [comment]
@ad96 Oh, yep, probably. I hadn't seen/used that feature before.

**Comment by ad96** [comment]
yep, @coblej showed me! :>

**Comment by ad96** [comment]
@seanaery you're welcome to accept the PR and squash it!

**Pull request merged**